R/usage_to_annual_nets_distributed.R

```diff
@@ -0,0 +1,234 @@
+#' Convert bednet usage to annual nets distributed per capita
+#' 
+#' Assumptions: 
+#' - Net use rates stays at 2019 levels for all countries
+#' - When converting from access to NPC, use the 2020 loess curve fit of access-NPC 
+#'   at the country-month level.
+#' 
+#' @param target_usage A single value or vector of desired target usages to model.
+#' @param country_iso3 Vector of ISO3 codes of African countries for which to return results. 
+#' Default = "all".
+#' @param extrapolate_npc Option to extrapolate target nets per capita beyond fitted 
+#' Loess curve. Default = "linear" for linear extrapolation. For any other inputs,
+#' NA outputs are returned for any target usages exceeding currently observed levels.
+#' @param k Fixed rate parameter of the internal function of net loss, used to 
+#' estimate the median net retention times by country. Default = 20 from
+#' Bertozzi-Villa, Amelia, et al. Nature communications 12.1 (2021): 1-12.
+#'
+#' @return Dataframe of access, nets per capita and annual nets distributed per capita
+#' for the desired steady-state target usage by country
+#' 
+#' @importFrom stats 'approx' 'reshape' 'loess' 'predict'
+#' @export
+
+find_annual_nets_distributed <- function(target_usage, country_iso3="all", 
+                                         extrapolate_npc = "linear", k=20) {
+  
+  ### 1. Convert target usage to nets per capita ###
+  
+  # Read in datasets of country-specific use rate and the Loess curve of 
+  # access vs nets per capita
+  data <- prepare_data()
+  country_use_rate <- data$use_rate_by_country
+  loess_for_prediction <- data$loess_for_prediction
+  
+  if(country_iso3[1]=="all") {
+    country_iso3 <- country_use_rate$iso3
+  }
+  
+  if(extrapolate_npc == "linear") {
+    # Add datapoint of (1,1) to allow linear extrapolation of nets per capita
+    # for access levels beyond those observed
+    loess_for_prediction <- rbind(loess_for_prediction,
+                                  data.frame(iso3=NA, month=NA, access=1, 
+                                             percapita_nets = 1, loess_predicted_access=1))
+    print("Access-nets per capita curve is extrapolated linearly beyond observed levels")
```

Comment:
As below comment (`warning()`)

Reply:
I did not change this to warning() since it is more of a reminder of what is inputted